srtcore/core.cpp

```diff
@@ -7697,9 +7734,26 @@ int CUDT::packLostData(CPacket& w_packet, steady_clock::time_point& w_origintime
         const int offset = CSeqNo::seqoff(m_iSndLastDataAck, w_packet.m_iSeqNo);
         if (offset < 0)
         {
-            LOGC(dlog.Error,
-                 log << "IPE: packLostData: LOST packet negative offset: seqoff(m_iSeqNo " << w_packet.m_iSeqNo
-                     << ", m_iSndLastDataAck " << m_iSndLastDataAck << ")=" << offset << ". Continue");
+            // XXX Likely that this will never be executed because if the upper
+            // sequence is not in the sender buffer, then most likely the loss 
+            // was completely ignored.
+            LOGC(dlog.Error, log << "IPE/EPE: packLostData: LOST packet negative offset: seqoff(m_iSeqNo "
+                << w_packet.m_iSeqNo << ", m_iSndLastDataAck " << m_iSndLastDataAck
+                << ")=" << offset << ". Continue");
+
+            // No matter whether this is right or not (maybe the attack case should be
+            // considered, and some LOSSREPORT flood prevention), send the drop request
+            // to the peer.
+            int32_t seqpair[2];
+            seqpair[0] = w_packet.m_iSeqNo;
+            seqpair[1] = m_iSndLastDataAck;
```

Comment:
const + inplace initialization?

Reply:
Not possible. The parameter in `sendCtrl` where `seqpair` is passed as non-const.

Whether it makes sense and should be there is another story. There's likely some refax ongoing that fixes the header of this function, or some underlying part, but that should be done separately.